### PR TITLE
912 sort print files for hand delivery

### PR DESCRIPTION
**Pull request by LukeLoze1**
Motivation and Context
Sort print files when pack codes for hand delivery

What has changed
When checking print files we take the actual print file rows and check that they're sorted as expected

How to test?
With printfilesvc from ticket

Links
https://trello.com/c/rZ6dszTh/912-sort-print-files-for-hand-delivery-13

### CODE REVIEW

File: acceptance_tests/features/steps/print_file.py [hunk 1]
```diff
@@ -188,6 +197,34 @@ def _get_print_file_rows_as_list(context, pack_code):
...
+        for csvrow in unsorted_actual_content_list
+    ]
+
+    # If called with an invalid packcode the test will fail, this is expected behaviour
+    sorted_list = []
+
+    # This will sort a list of lists based on template ICL or QM
+    # in both cases we're sorting by field_officer_id and org_name
+    if pack_code in ICL_PACKCODES_SORTED_BY_PRODUCTION_CODE:
+        sorted_list = sorted(split_csv_rows, key=lambda row: (row[14], row[12]))
```
Comment (NickGrantX) [inline acceptance_tests/features/steps/print_file.py]: do you think these 'magic' numbers should be constants?

File: acceptance_tests/features/steps/print_file.py [hunk 2]
```diff
@@ -188,6 +197,34 @@ def _get_print_file_rows_as_list(context, pack_code):
...
+    elif pack_code in QM_PACKCODES_SORTED_BY_PRODUCTION_CODE:
+        sorted_list = sorted(split_csv_rows, key=lambda x: (x[15], x[14]))
```
Comment (NickGrantX) [inline acceptance_tests/features/steps/print_file.py]: do you think these 'magic' numbers should be constants?

Comment (NickGrantX) [reply acceptance_tests/features/steps/print_file.py]: also, `x` is not a brilliant variable name

**Review by NickGrantX — CHANGES_REQUESTED** [review]
Might be nice to see sorting by both field office & org... I feel like there aren't enough rows with the same field officer to be sure that the org sorting is working.

File: acceptance_tests/features/steps/print_file.py [hunk 3]
```diff
@@ -22,6 +23,14 @@
 logger = wrap_logger(logging.getLogger(__name__))
 
 
+ICL_PACKCODES_SORTED_BY_PRODUCTION_CODE = [
+    'D_CE1A_ICLCR1', 'D_CE1A_ICLCR2B', 'D_ICA_ICLR1', 'D_ICA_ICLR1',
+    'D_ICA_ICLR2B', 'D_ICA_ICLR2B', 'D_CE4A_ICLR4', 'D_CE4A_ICLS4'
+]
+
+QM_PACKCODES_SORTED_BY_PRODUCTION_CODE = ['D_FDCE_I1', 'D_FDCE_I2', 'D_FDCE_I4', 'D_FDCE_H1', 'D_FDCE_H2']
```
Comment (AdamHawtin) [inline acceptance_tests/features/steps/print_file.py]: The names `X_PACKCODES_SORTED_BY_PRODUCTION_CODE` threw me for a minute, it sounds like it's a list of packcodes which has been sorted on production code, and I was trying to remember what a `production_code` was 🤔. Maybe `X_PACKCODES_WHICH_ARE_SORTED` instead?

**Review by AdamHawtin — CHANGES_REQUESTED** [review]

File: acceptance_tests/features/steps/print_file.py (hunk 2 above)
Comment (LukeLoze1) [reply acceptance_tests/features/steps/print_file.py]: doh missed a row, literally

**Review by AdamHawtin — APPROVED** [review]

**Review by NickGrantX — APPROVED** [review]
LGTM 👍

**Pull request merged**